fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneQueryTest.java

```diff
@@ -905,4 +906,39 @@ void covering() throws Exception {
         }
     }
 
+    @Test
+    void fullGroupScan() throws Exception {
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context, md -> {
+                md.removeIndex(MAP_AND_FIELD_ON_LUCENE_INDEX.getName());
+            }, SIMPLE_TEXT_SUFFIXES);
+            QueryComponent groupFilter = Query.field("entry").oneOfThem().matches(Query.field("key").equalsValue("a"));
+            RecordQuery query = RecordQuery.newBuilder()
+                    .setRecordType(MAP_DOC)
+                    .setFilter(groupFilter)
+                    .build();
+            RecordQueryPlan plan = planner.plan(query);
+            Matcher<RecordQueryPlan> matcher = filter(groupFilter, typeFilter(equalTo(Collections.singleton(TextIndexTestUtils.MAP_DOC)), scan(unbounded())));
+            assertThat(plan, matcher);
+        }
+
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context, md -> {
+                md.removeIndex(MAP_AND_FIELD_ON_LUCENE_INDEX.getName());
+                md.removeIndex(MAP_ON_LUCENE_INDEX.getName());
+                md.addIndex(MAP_DOC, new Index("GroupedMap", new GroupingKeyExpression(concat(field("group"), mainExpression), 1), LuceneIndexTypes.LUCENE));
+            }, SIMPLE_TEXT_SUFFIXES);
+            QueryComponent groupFilter = Query.and(
+                    Query.field("group").equalsValue(1L),
+                    Query.field("entry").oneOfThem().matches(Query.field("key").equalsValue("a")));
+            RecordQuery query = RecordQuery.newBuilder()
+                    .setRecordType(MAP_DOC)
+                    .setFilter(groupFilter)
+                    .build();
+            RecordQueryPlan plan = planner.plan(query);
+            Matcher<RecordQueryPlan> matcher = filter(groupFilter, typeFilter(equalTo(Collections.singleton(TextIndexTestUtils.MAP_DOC)), scan(unbounded())));
+            assertThat(plan, matcher);
```

Comment:
Is there a test showing that the plan actually executes correctly (that is, that it returns all results and doesn't trip over the lack of bounds)?

Reply:
I think the reasoning was that this need only check that a non-Lucene plan is generated and that such tests would exist in the core.

Reply:
Oh, I see--the plan in question is a full store scan, not an unbounded scan of the index. Got it